jsapp/js/assetUtils.ts

```diff
@@ -394,45 +396,63 @@ export function isRowSpecialLabelHolder(mainRow, holderRow) {
 
 /**
  * An internal helper function for DRY code
- * @param {object} row
- * @param {number} index
- * @returns {string} label
  */
-function getRowLabelAtIndex(row, index) {
+function getRowLabelAtIndex(
+  row: SurveyRow | SurveyChoice,
+  index: number
+): string | null {
   if (Array.isArray(row.label)) {
-    return row.label[index];
+    return row.label[index] || null;
   } else {
-    return row.label;
+    return row.label || null;
   }
 }
 
-/**
- * @param {string} type - one of QUESTION_TYPES
- * @returns {Node|null}
- */
-export function renderQuestionTypeIcon(type) {
-  let typeDef;
-  if (type === SCORE_ROW_TYPE) {
-    typeDef = QUESTION_TYPES.score;
-  } else if (type === RANK_LEVEL_TYPE) {
-    typeDef = QUESTION_TYPES.rank;
-  } else {
-    typeDef = QUESTION_TYPES[type];
+export function renderQuestionTypeIcon(
+  rowType: AnyRowTypeName
+): React.DetailedReactHTMLElement<{}, HTMLElement> | null {
```

Comment:
Still new to some syntax, so I want to make sure I understand this generic correctly:
` React.DetailedReactHTMLElement<{}, HTMLElement>` is an object that takes in an HTML element and some object as parameters for that HTML element?

Reply:
I think the object is for custom node attributes. I don't fully get the `<>` in types yet. I'm often just looking out what people do and looking through the types definitions of our packages :)